sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -281,18 +408,27 @@ def _check_categories(self, X):
 
         for f_idx in range(n_features):
             if is_categorical[f_idx]:
-                categories = np.unique(X[:, f_idx])
-                missing = np.isnan(categories)
-                if missing.any():
-                    categories = categories[~missing]
-
-                # Treat negative values for categorical features as missing values.
-                negative_categories = categories < 0
-                if negative_categories.any():
-                    categories = categories[~negative_categories]
-
-                if hasattr(self, "feature_names_in_"):
-                    feature_name = f"'{self.feature_names_in_[f_idx]}'"
+                if use_pandas_categorical:
+                    # pandas categories do not include missing values so there is
+                    # no need to filter them out.
+                    categories = X.iloc[:, f_idx].unique().dropna().to_numpy()
+                    # OrdinalEncoder requires categories backed by numerical values
+                    # to be sorted
+                    if categories.dtype.kind not in "OUS":
+                        categories = np.sort(categories)
```

Comment:
Do you think it could be possible to get a specific test for this. I assume that we test it indirectly because codecov does not complain that this line is not covered but it could be cool to have an explicit test.

Reply:
In https://github.com/scikit-learn/scikit-learn/pull/26411/commits/5fea2ddd225c1692709a1f88e3a1d41f7c9dcb0f I updated the test to directly check the categories.